CRM/Activity/BAO/Activity.php

```diff
@@ -2562,27 +2562,30 @@ public static function getContactActivitySelector(&$params) {
           $activity['target_contact_name'] = $values['recipients'];
         }
         elseif (isset($values['target_contact_count']) && $values['target_contact_count']) {
-          $activity['target_contact_name'] = '';
           $firstTargetName = reset($values['target_contact_name']);
           $firstTargetContactID = key($values['target_contact_name']);
 
-          $targetLink = CRM_Utils_System::href($firstTargetName, 'civicrm/contact/view', "reset=1&cid={$firstTargetContactID}");
-          if ($showContactOverlay) {
-            $targetTypeImage = CRM_Contact_BAO_Contact_Utils::getImage(
-              CRM_Contact_BAO_Contact::getContactType($firstTargetContactID),
-              FALSE,
-              $firstTargetContactID);
-            $activity['target_contact_name'] .= "<div>$targetTypeImage  $targetLink";
-          }
-          else {
-            $activity['target_contact_name'] .= $targetLink;
-          }
+          // The first target may not be accessable to the logged in user dev/core#1052
+          if ($firstTargetName) {
+            $activity['target_contact_name'] = '';
```

Comment:
@seamuslee001 so my worry is this line - it looks to me that you could get leakage here - ie if we have a target name for one activity & not the next we are not blanking it outside the if statement so it could bleed

Reply:
@eileenmcnaughton so would you prefer that was on the outside of the if then?

Reply:
@seamuslee001 right - my read is that by having it inside the if we could get leakage between the foreach iterations